Allow ext/mssql and ext/pdo_dblib to build with >=freetds-0.82
Original patch by jklowden at freetds dot org
ext/pdo_dblib patch by matthias at dsx ado at
Rediffed for Gentoo (wrong direction)
http://bugs.php.net/bug.php?id=44991

diff -r 57e37b68a255 -r f1222f67e562 ext/mssql/config.m4
--- a/ext/mssql/config.m4	Tue Jul 01 19:37:24 2008 +0200
+++ b/ext/mssql/config.m4	Thu Jul 17 11:35:57 2008 +0200
@@ -10,11 +10,11 @@
 
   if test "$PHP_MSSQL" = "yes"; then
     for i in /usr/local /usr; do
-      if test -f $i/include/tds.h; then
+      if test -f $i/include/sybdb.h; then
         FREETDS_INSTALLATION_DIR=$i
         FREETDS_INCLUDE_DIR=$i/include
         break
-      elif test -f $i/include/freetds/tds.h; then
+      elif test -f $i/include/freetds/sybdb.h; then
         FREETDS_INSTALLATION_DIR=$i
         FREETDS_INCLUDE_DIR=$i/include/freetds
         break
@@ -27,10 +27,10 @@
 
   elif test "$PHP_MSSQL" != "no"; then
 
-    if test -f $PHP_MSSQL/include/tds.h; then
+    if test -f $PHP_MSSQL/include/sybdb.h; then
       FREETDS_INSTALLATION_DIR=$PHP_MSSQL
       FREETDS_INCLUDE_DIR=$PHP_MSSQL/include
-    elif test -f $PHP_MSSQL/include/freetds/tds.h; then
+    elif test -f $PHP_MSSQL/include/freetds/sybdb.h; then
       FREETDS_INSTALLATION_DIR=$PHP_MSSQL
       FREETDS_INCLUDE_DIR=$PHP_MSSQL/include/freetds
     else
@@ -38,8 +38,8 @@
     fi
   fi  
 
-  if test ! -r "$FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libtds.a" && test ! -r "$FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libtds.so"; then
-     AC_MSG_ERROR(Could not find $FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libtds.[a|so])
+  if test ! -r "$FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libsybdb.a" && test ! -r "$FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libsybdb.so"; then
+     AC_MSG_ERROR(Could not find $FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libsybdb.[a|so])
   fi
 
   PHP_ADD_INCLUDE($FREETDS_INCLUDE_DIR)
diff -r 57e37b68a255 -r f1222f67e562 ext/pdo_dblib/config.m4
--- a/ext/pdo_dblib/config.m4	Tue Jul 01 19:37:24 2008 +0200
+++ b/ext/pdo_dblib/config.m4	Thu Jul 17 11:35:57 2008 +0200
@@ -13,11 +13,11 @@
   if test "$PHP_PDO_DBLIB" = "yes"; then
 
     for i in /usr/local /usr; do
-      if test -f $i/include/tds.h; then
+      if test -f $i/include/sybdb.h; then
         PDO_FREETDS_INSTALLATION_DIR=$i
         PDO_FREETDS_INCLUDE_DIR=$i/include
         break
-      elif test -f $i/include/freetds/tds.h; then
+      elif test -f $i/include/freetds/sybdb.h; then
         PDO_FREETDS_INSTALLATION_DIR=$i
         PDO_FREETDS_INCLUDE_DIR=$i/include/freetds
         break;
@@ -30,10 +30,10 @@
 
   elif test "$PHP_PDO_DBLIB" != "no"; then
 
-    if test -f $PHP_PDO_DBLIB/include/tds.h; then
+    if test -f $PHP_PDO_DBLIB/include/sybdb.h; then
       PDO_FREETDS_INSTALLATION_DIR=$PHP_PDO_DBLIB
       PDO_FREETDS_INCLUDE_DIR=$PHP_PDO_DBLIB/include
-    elif test -f $PHP_PDO_DBLIB/include/freetds/tds.h; then
+    elif test -f $PHP_PDO_DBLIB/include/freetds/sybdb.h; then
       PDO_FREETDS_INSTALLATION_DIR=$PHP_PDO_DBLIB
       PDO_FREETDS_INCLUDE_DIR=$PHP_PDO_DBLIB/include/freetds
     else
@@ -45,8 +45,8 @@
     PHP_LIBDIR=lib
   fi
 
-  if test ! -r "$PDO_FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libtds.a" && test ! -r "$PDO_FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libtds.so"; then
-     AC_MSG_ERROR(Could not find $PDO_FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libtds.[a|so])
+  if test ! -r "$PDO_FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libsybdb.a" && test ! -r "$PDO_FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libsybdb.so"; then
+     AC_MSG_ERROR(Could not find $PDO_FREETDS_INSTALLATION_DIR/$PHP_LIBDIR/libsybdb.[a|so])
   fi
 
   PHP_ADD_INCLUDE($PDO_FREETDS_INCLUDE_DIR)
